privacyidea/lib/policy.py

```diff
@@ -588,6 +590,33 @@ def get_policies(self, name=None, scope=None, realm=None, active=None,
 
         return reduced_policies
 
+    @staticmethod
+    def choose_prioritized_policy(policies):
+        """
+        Given a list of policy dictionaries, find the policy with the highest priority,
+        i.e. the policy with the smallest 'priority' value.
+
+        In case the list is empty, this returns the empty list.
+        Otherwise, this returns a list with one element, the prioritized policy.
+
+        In case there are multiple policies with the same priority, this raises
+        a PolicyError.
+
+        :param policies: list of dictionaries
+        :return: a list
+        """
+        if policies:
+            priorities = [p['priority'] for p in policies]
+            total_priorities = len(priorities)
+            distinct_priorities = len(set(priorities))
+            if total_priorities != distinct_priorities:
+                raise PolicyError(u"There are policies with conflicting priorities")
```

Comment:
I think this might often be a non-expected result. If there are two policies, with the same priority and the same resulting value - why should this raise an error?

The admin would argue, that the configuration create a clear logic with no contradiction. So he probalby would expect privacyIDEA to act accordingly.